cogs/moderation.py

```diff
@@ -1,29 +1,57 @@
 import asyncio
+from collections import Counter
+import typing as t
+from .utils.formats import plural
 import textwrap
 
 import discord
 from discord import Color, Embed
 from discord.ext.commands import (
+    BadArgument,
     Bot,
     Cog,
     Context,
+    Converter,
     bot_has_permissions,
     command,
+    guild_only,
+    Greedy,
     has_permissions,
+    MemberConverter,
 )
 
 
-class Moderation(Cog):
-    """This cog provides moderation commands."""
+def can_execute_action(ctx: Context, user: discord.User, target: discord.User) -> bool:
+    return user.id == ctx.bot.owner_id or user == ctx.guild.owner or user.top_role > target.top_role
+
+
+class MemberNotFound(Exception):
+    pass
+
+
+async def resolve_member(self, guild: discord.Guild, member_id: int) -> discord.User:
+    user = guild.get_member(member_id)
+    if user is None:
+        if guild.chunked:
+            raise MemberNotFound()
+        try:
+            user = await guild.fetch_member(member_id)
+        except discord.NotFound:
+            raise MemberNotFound()
+    return user
 
-    def __init__(self, bot: Bot) -> None:
-        self.bot = bot
+
+# TODO : Clean everything up
+class Moderation(Cog):
+    def __init__(self, client: discord.Bot) -> None:
+        self.client = client
```

Comment:
According to standards, we should use `bot` not `client`:
```suggestion
    def __init__(self, bot: Bot) -> None:
        self.bot = bot
```
Also, why were you type hinting this with `discord.Bot` when you can just do `Bot`, you have it imported